docs/RFCS/txn_gc.md

```diff
@@ -0,0 +1,62 @@
+- Feature Name: txn-gc
+- Status: draft
+- Start Date: 2015-11-11
+- RFC PR:
+- Cockroach Issue:
+
+# Summary
+
+* rename `{Response->Sequence}Cache`.
```

Comment:
While we're renaming things, it's not really a "cache". I'm not sure what to call it, though. If "table" didn't have a special meaning at the structured/sql layer I'd call it `TxnSequenceTable`.